### ISSUE TITLE
Swiping on Calendar composable results in crash with following exception "java.lang.AbstractMethodError: abstract method "float androidx.compose.foundation.gestures.snapping.SnapLayoutInfoProvider.calculateApproachOffset(float)""

### ISSUE THREAD

**Issue by RickertBrandsen** [kizitonwose/Calendar#511]
### Library information:
 - Version: 2.4.0
 - Compose module

### Describe the bug**
Created new clean android studio project couple of times and took the compose sample number 3 and everything ran smoothly without any issues. But once I imported in my larger project the following exception when swiping on calendar appeared resulting in app crash. 

```
java.lang.AbstractMethodError: abstract method "float androidx.compose.foundation.gestures.snapping.SnapLayoutInfoProvider.calculateApproachOffset(float)"
at androidx.compose.foundation.gestures.snapping.SnapFlingBehavior$fling$result$1.invokeSuspend(SnapFlingBehavior.kt:167)
at androidx.compose.foundation.gestures.snapping.SnapFlingBehavior$fling$result$1.invoke(Unknown Source:8)
at androidx.compose.foundation.gestures.snapping.SnapFlingBehavior$fling$result$1.invoke(Unknown Source:4)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:167)
at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
at androidx.compose.foundation.gestures.snapping.SnapFlingBehavior.fling(SnapFlingBehavior.kt:165)
at androidx.compose.foundation.gestures.snapping.SnapFlingBehavior.performFling(SnapFlingBehavior.kt:151)
at androidx.compose.foundation.gestures.snapping.SnapFlingBehavior.performFling(SnapFlingBehavior.kt:130)
at androidx.compose.foundation.gestures.ScrollingLogic$doFlingAnimation$2.invokeSuspend(Scrollable.kt:844)
at androidx.compose.foundation.gestures.ScrollingLogic$doFlingAnimation$2.invoke(Unknown Source:8)
at androidx.compose.foundation.gestures.ScrollingLogic$doFlingAnimation$2.invoke(Unknown Source:4)
at androidx.compose.foundation.gestures.DefaultScrollableState$scroll$2$1.invokeSuspend(ScrollableState.kt:181)
at androidx.compose.foundation.gestures.DefaultScrollableState$scroll$2$1.invoke(Unknown Source:8)
at androidx.compose.foundation.gestures.DefaultScrollableState$scroll$2$1.invoke(Unknown Source:4)
at androidx.compose.foundation.MutatorMutex$mutateWith$2.invokeSuspend(MutatorMutex.kt:173)
at androidx.compose.foundation.MutatorMutex$mutateWith$2.invoke(Unknown Source:8)
at androidx.compose.foundation.MutatorMutex$mutateWith$2.invoke(Unknown Source:4)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
at androidx.compose.foundation.MutatorMutex.mutateWith(MutatorMutex.kt:166)
at androidx.compose.foundation.gestures.DefaultScrollableState$scroll$2.invokeSuspend(ScrollableState.kt:178)
at androidx.compose.foundation.gestures.DefaultScrollableState$scroll$2.invoke(Unknown Source:8)
at androidx.compose.foundation.gestures.DefaultScrollableState$scroll$2.invoke(Unknown Source:4)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
at androidx.compose.foundation.gestures.DefaultScrollableState.scroll(ScrollableState.kt:177)
at androidx.compose.foundation.lazy.LazyListState.scroll(LazyListState.kt:295)
at androidx.compose.foundation.gestures.ScrollableState.scroll$default(ScrollableState.kt:53)
at androidx.compose.foundation.gestures.ScrollingLogic.doFlingAnimation-QWom1Mo(Scrollable.kt:831)
at androidx.compose.foundation.gestures.ScrollingLogic$onDragStopped$performFling$1.invokeSuspend(Scrollable.kt:807)
at androidx.compose.foundation.gestures.ScrollingLogic$onDragStopped$performFling$1.invoke-sF-c-tU(Unknown Source:12)
```

Initially thought, that it might be crashing because my composable being wrapped inside composable's Constraint layout, but it is actually crashing in general, but as I said, strange thing is, that in plain projects was everything working fine for me.

Is there anything I am overlooking? Can post my configs/etc if needed. Thanks for eventual helping hand.

### To Reproduce (if applicable)
1. Go to Calendar screen
2. Swipe on Calendar
4. See app crashing and Exception in logs

### Expected behavior (if applicable)
App should not be crashing in this case.

**Comment by kizitonwose** [comment]
Can you verify that you are using the appropriate calendar version? 

<img width="307" alt="Screenshot 2023-11-26 at 09 16 44" src="https://github.com/kizitonwose/Calendar/assets/15170090/93d0a1eb-0a7d-48a8-9c33-b2111215ca69">

This was fixed in #501

**Comment by RickertBrandsen** [comment]
> Can you verify that you are using the appropriate calendar version?
> 
> <img alt="Screenshot 2023-11-26 at 09 16 44" width="307" src="https://private-user-images.githubusercontent.com/15170090/285637644-93d0a1eb-0a7d-48a8-9c33-b2111215ca69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDA5OTY0MzMsIm5iZiI6MTcwMDk5NjEzMywicGF0aCI6Ii8xNTE3MDA5MC8yODU2Mzc2NDQtOTNkMGExZWItMGE3ZC00OGE4LTljMzMtYjIxMTEyMTVjYTY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI2VDEwNTUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwMTZiOWZlZjIxYjgwNGVhNTQxMTVjNDJlODI1NzU1YWJlYzUyMmEwMjg2MzgyOTIxYWM4MjQ5ODRmOWM2ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h1VtFUA3DKwqIHTlqqh_FFVzY8XBi56ujwuQYvaYQnE">
> This was fixed in #501

Yes, you are right, have been on wrong versions, but thanks for mentioning the #501.
